src/common/compression.c

```diff
@@ -135,3 +119,79 @@ daos_str2compresscontprop(const char *value)
 
 	return -DER_INVAL;
 }
+
+/**
+ * struct daos_compressor functions
+ */
+
+int
+daos_compressor_init(struct daos_compressor **obj, struct compress_ft *ft)
+{
+    struct daos_compressor    *result;
+    int             rc = 0;
+
+    if (!ft) {
+ 		D_ERROR("No function table");
```

Comment:
(style)  code indent should use tabs where possible
(style)  please, no space before tabs